OpenSim/Simulation/Model/ModelComponent.cpp

```diff
@@ -127,4 +127,16 @@ void ModelComponent::updateFromXMLNode(SimTK::Xml::Element& aNode,
     Super::updateFromXMLNode(aNode, versionNumber);
 }
 
+const SimTK::Vec3 ModelComponent::
+getScaleFactors(const ScaleSet& scaleSet, const Frame& frame) const
```

Comment:
:+1:  I like this because it hides the implementation of how you get scale factors given a `Frame`. __In the future__, we may consider passing `ModelComponent` instead of `Frame` since we are moving towards the `ModelComponent` knowing how to scale itself, so it should be able to access scale factors to do that.

Reply:
That sounds good. We might also consider using something more general than ScaleSet so "pre-scale lengths" and other scaling-related cache can be stored there instead of in private member variables like `GeometryPath::_preScaleLength`. Opened #1996 to track.

Reply:
(We wouldn't need to *pass in* a reference to the ModelComponent since this is a method on the ModelComponent class. But, yes, each ModelComponent could implement its own version of `getScaleFactors()`).